test/runtests.jl

```diff
@@ -15,85 +14,99 @@ const rng = StableRNG(12345)
         # Convolution
         @test WeightInitializers._nfan(4, 5, 6) == 4 .* (5, 6)
     end
+    @testset "rng = $(typeof(rng))" for rng in [StableRNG(12345), Random.default_rng(),
+        CUDA.default_rng(), CURAND.default_rng(),
+    ]
+        @testset "Sizes and Types: $init" for init in [zeros32, ones32, rand32, randn32,
+            kaiming_uniform, kaiming_normal, glorot_uniform, glorot_normal, truncated_normal,
+        ]
+            # Sizes
+            @test size(init(3)) == (3,)
+            @test size(init(rng, 3)) == (3,)
+            @test size(init(3, 4)) == (3, 4)
+            @test size(init(rng, 3, 4)) == (3, 4)
+            @test size(init(3, 4, 5)) == (3, 4, 5)
+            @test size(init(rng, 3, 4, 5)) == (3, 4, 5)
+            # Type
+            @test eltype(init(rng, 4, 2)) == Float32
+            @test eltype(init(4, 2)) == Float32
+            # RNG Closure
+            cl = init(rng)
+            @test cl(3) isa AbstractArray{Float32, 1}
+            @test cl(3, 5) isa AbstractArray{Float32, 2}
```

Comment:
Don't check `AbstractArray`. Have a list mapping the rngtype to the correct array type, and check against that. For ex, CURAND and CUDA should produce CuArray while the others need to produce Array